beacon/manager.go

```diff
@@ -251,22 +254,23 @@ func (s *Manager) RunPayloadAttributesSubscription(ctx context.Context, state St
 	client.SubscribeToPayloadAttributesEvents(c)
 
 	for payloadAttributes := range c {
-		logger = logger.WithField("slot", payloadAttributes.Data.ProposalSlot-1)
+		slot := payloadAttributes.Data.ProposalSlot - 1
+		logger = logger.WithField("slot", slot)
 
-		proposalSlot := payloadAttributes.Data.ProposalSlot - 1
-		if slot := state.HeadSlotPayloadAttributes(); proposalSlot <= slot {
+		paHeadSlot, ok := state.SetHeadSlotPayloadAttributesIfHigher(slot)
+		if !ok {
+			logger.WithField("slotHead", paHeadSlot).Warn("received old payload attributes")
 			continue
 		}
-		state.SetHeadSlotPayloadAttributes(proposalSlot)
 
 		headSlot := state.HeadSlot()
 
-		if proposalSlot < uint64(headSlot) {
+		if slot <= uint64(headSlot)-NumberOfSlotsInState {
```

Comment:
Why is this? Don't we just want to check it against the head slot?

Reply:
Nope, we want to allow a bit older (we keep state of last `NumberOfSlotsInState` slots), in case there are old submissions